src/Features/CSharp/Portable/SplitStringLiteral/StringSplitter.cs

```diff
@@ -47,40 +44,45 @@ internal abstract partial class StringSplitter
             UseTabs = useTabs;
             TabSize = tabSize;
             _indentStyle = indentStyle;
-            CancellationToken = cancellationToken;
         }
 
-        public static StringSplitter TryCreate(
+        public static async Task<(Document? document, int caretPosition)> TrySplitAsync(
             Document document, int position,
             bool useTabs, int tabSize, FormattingOptions.IndentStyle indentStyle,
             CancellationToken cancellationToken)
         {
-            var root = document.GetSyntaxRootSynchronously(cancellationToken);
-            var sourceText = root.SyntaxTree.GetText(cancellationToken);
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
this is a regression.  Please still call GetSyntaxRootSynchronously (which can actually do everything needed on the calling thread).  The reason for this is that all these operations need to operate as if they were synchonrous so they can finish as quickly as possible asn relinquish the UI thread that is blocking, to have hte least impact on typing.  By explicitly awaiting (esp. with CA(false)) we are now dependent on tasks completing on the threadpool, which may be occupied with other work.   As there is no way for us to prioritize threadpool work, this effectively makes the UI thread (in a typing scenario) blocked on arbitrary bg work), something we never really want to happen (unless there is a TWD up).  And, because this is pure typing, we do not ever even want a TWD tehre.

This is a tough space.  However, the current design was such to balance these concerns.  GetSyntaxRootSynchronously exists as an API for precisely this purpose.  So we can have smart syntactic features, while still operating efficiently within the typing loop without having to hit the threadpool.

Reply:
paging @jasonmalinowski  as well

Reply:
OK, but there are other async calls that we are waiting on. Also if this is move to LSP we will have to use async.

Reply:
[IIndentationServiceExtensions.GetIndentation](https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Indentation/IIndentationService.cs,60) calls `document.GetOptionsAsync(cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken)`. 

[AbstractIndentationService.GetIndenter](https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Indentation/AbstractIndentationService.cs,53) also calls a couple of async methods and waits for the results.

Reply:
We can make this entirely synchronous but I think to do that we need new editor API: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1429383
This would give us access to `ICodingConventionsSnapshot` representing editorconfig options that the editor read that corresponds to a given `ITextSnapshot`. That way we can avoid calling `GetOptionsAsync`.
Also, if this needs to stay in-proc for perf reasons we should avoid using `Document` in this code so that we can avoid populating `Solution` snapshot in-proc at some point.